app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -86,15 +90,23 @@ def save_inventory_single(type, parent, hash, child_keys = [], extra_keys = [],
     save_child_inventory(child, hash, child_keys)
   end
 
-  def save_inventory_with_findkey(association, hash, deletes, new_records, record_index)
+  def save_inventory_with_findkey(association, hash, deletes, new_records, record_index,
+                                  reconnect_index: nil, reconnect_records: nil)
     # Find the record, and update if found, else create it
     found = record_index.fetch(hash)
-    if found.nil?
-      found = association.build(hash.except(:id))
-      new_records << found
-    else
+    if found
       found.update_attributes!(hash.except(:id, :type))
       deletes.delete(found) unless deletes.blank?
+    else
+      found = reconnect_index.fetch(hash) if reconnect_index
+      if found
+        found.update_attributes!(hash.except(:id, :type))
+        deletes.delete(found) unless deletes.blank?
```

Comment:
I think so too, but wasn't sure if `deletes` may ever be larger than `association`...